File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> s;

        for (int i = 0; i < size(nums); i++) {
            if (s.count(nums[i]) > 0) {
                return true;
            }
            else {
                s.insert(nums[i]);
            }
        }

        return false;

    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        map = {}

        for i, num in enumerate(nums):
            x = target - num

            if x in map:
                return [map[x], i]

            else:
                map[num] = i

        return False